src/Widgets/SearchToolbar.vala

```diff
@@ -1,101 +1,86 @@
-            i.pixel_size = 16;
-            var next_button = new Gtk.ToolButton (i, null);
-            next_button.set_tooltip_text (_("Next result"));
+            var next_button = new Gtk.ToolButton (new Gtk.Image.from_icon_name ("go-down-symbolic", Gtk.IconSize.SMALL_TOOLBAR), null);
+            next_button.tooltip_text = _("Next result");
 
-            this.add (tool_search_entry);
-            this.add (previous_button);
-            this.add (next_button);
+            add (tool_search_entry);
+            add (previous_button);
+            add (next_button);
 
-            this.show_all ();
-            this.set_style (Gtk.ToolbarStyle.ICONS);
-            this.get_style_context ().add_class ("search-bar");
+            get_style_context ().add_class ("search-bar");
+            show_all ();
 
-            // Signals and callbacks
-            this.clear.connect (clear_cb);
-            this.grab_focus.connect (grab_focus_cb);
-            this.search_entry.search_changed.connect (search_changed_cb);
-            previous_button.clicked.connect (previous_search);
-            next_button.clicked.connect (next_search);
-        }
+            grab_focus.connect (() => {
+                search_entry.grab_focus ();
+            });
 
-        void clear_cb () {
-            this.search_entry.text = "";
-        }
+            search_entry.search_changed.connect (() => {
+                try {
+                    // FIXME Have a configuration menu or something.
+                    var regex = new Regex (Regex.escape_string (search_entry.text), RegexCompileFlags.CASELESS);
+                    window.current_terminal.search_set_gregex (regex, 0);
+                    window.current_terminal.search_set_wrap_around (true);
+                } catch (RegexError er) {
+                    warning ("There was an error to compile the regex: %s", er.message);
+                }
+            });
 
-        void grab_focus_cb () {
-            this.search_entry.grab_focus ();
+            previous_button.clicked.connect (previous_search);
+            next_button.clicked.connect (next_search);
         }
 
-        void search_changed_cb () {
-            try {
-                // FIXME Have a configuration menu or something.
-                var regex = new Regex (Regex.escape_string (search_entry.text),
-                                       RegexCompileFlags.CASELESS);
-                this.window.current_terminal.search_set_gregex (regex, 0);
-                this.window.current_terminal.search_set_wrap_around (true);
-            } catch (RegexError er) {
-                warning ("There was an error to compile the regex: %s", er.message);
-            }
+        public void clear () {
+            search_entry.text = "";
         }
 
         public void previous_search () {
-            this.window.current_terminal.search_find_previous ();
+            window.current_terminal.search_find_previous ();
```

Comment:
Is there a reason why the handler for search_changed, which is pretty 7 lines long was moved to an anonymous callback, but these ones which are 1 line long haven't been?